plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -271,6 +273,9 @@ open class KtlintPlugin : Plugin<Project> {
         android.set(extension.android)
         ignoreFailures.set(extension.ignoreFailures)
         outputToConsole.set(extension.outputToConsole)
+        if (!target.isConsolePlain()) {
+            coloredOutput.set(extension.coloredOutput)
```

Comment:
You should probably do this in a `extension.coloredOutput.map { }` where you put in the debug log that you are disabling the user choice because of the `isConsolePlain()` flag.